code/src/test/kotlin/mocks/MockTransport.kt

```diff
@@ -0,0 +1,52 @@
+package mocks
+
+import com.google.api.client.http.LowLevelHttpRequest
+import com.google.api.client.testing.http.MockHttpTransport
+import com.google.api.client.testing.http.MockLowLevelHttpRequest
+import com.google.api.client.testing.http.MockLowLevelHttpResponse
+import java.util.Queue
+
+class MockTransport(
+    private val authResponse: MockLowLevelHttpResponse,
+    private val apiResponse: MockLowLevelHttpResponse
+): MockHttpTransport() {
+    override fun buildRequest(method: String?, url: String?): LowLevelHttpRequest {
+        requireNotNull(url)
+        requireNotNull(method)
+
+        val request: MockLowLevelHttpRequest = super.buildRequest(method, url) as MockLowLevelHttpRequest
+
+        request.response = if (request.url.equals(Endpoint.AUTHENTICATION_ENDPOINT)) {
+            apiResponse
+        } else {
+            apiResponse
+        }
```

Comment:
```suggestion
        request.response = apiResponse
```

Reply:
oh, is the top one meant to be `authResponse`?

Reply:
fixed.